torch/_inductor/codegen/cpp.py

```diff
@@ -2585,28 +2587,35 @@ def reduction_combine_vec(
             raise NotImplementedError
 
     def indirect_assert(self, var, lower, upper, mask=None):
-        assert not mask, "do not support mask in indirect_indexing assertion"
         assert isinstance(var, CppCSEVariable)
         assert var.dtype is not None
-        if not var.is_vec:
+        if not (var.is_vec or (mask and mask.is_vec)):
```

Comment:
Does the existing test cover the this - either `var` or `mask` is vec?

Reply:
The new test does cover when `var` is not vec, but `mask` is.